go/carmen/memory_footprint.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package carmen
+
+import "github.com/Fantom-foundation/Carmen/go/common"
+
+type MemoryFootprint interface {
```

Comment:
This interface should also have its `memory_footprint_test.go`, even if its implementation is trivial.

Reply:
Are tests really necessary? The tests would only test already tested funcions since the interface `MemoryFootprint` only does not add any logic.

Reply:
it is the ideal situation to test each layer independently because the code can move around in the future and the dependencies between layers may change and the path that is accidentally tested now from another layer can get lost. 

We do not have this ideal situation everywhere at the moment, but we aim at improving the code all the time, and especially here it is really a few lines of code, i.e. a very low effort needed to reach this ideal state.